src/VisualStudio/IntegrationTest/TestUtilities/Common/Reference.cs

```diff
@@ -0,0 +1,45 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using Roslyn.Utilities;
+
+namespace Microsoft.VisualStudio.IntegrationTest.Utilities.Common
+{
+    [Serializable]
+    public class Reference : IEquatable<Reference>
+    {
+        public string FilePath { get; set; }
+        public int Line { get; set; }
+        public int Column { get; set; }
+        public string Code { get; set; }
+
+        public Reference() { }
+
+        public bool Equals(Reference other)
+        {
+            if (other == null)
+            {
+                return false;
+            }
+
+            return FilePath.Equals(other.FilePath, StringComparison.OrdinalIgnoreCase)
+                && Line == other.Line
+                && Column == other.Column
+                && Code.Equals(other.Code);
+        }
+
+        public override bool Equals(object obj)
+            => Equals(obj as Reference);
+
+        public override int GetHashCode()
+        {
+            return Hash.Combine(FilePath,
```

Comment:
nit: We really should have an overload that just takes a params object[], it would make this so much more readable.